exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/FunctionGenerationHelper.java

```diff
@@ -127,7 +133,7 @@ protected static boolean isComparableType(MajorType type) {
 
   private static String formatCanNotCompareMsg(MajorType left, MajorType right) {
     StringBuilder sb = new StringBuilder();
-    sb.append("Map, Array or repeated scalar type should not be used in group by, order by or in a comparison operator. Drill does not support compare between ");
+    sb.append("Map, Array, Union or repeated scalar type should not be used in group by, order by or in a comparison operator. Drill does not support compare between ");
```

Comment:
Yes, they could be comparable, but that functionality has not been implemented yet.